File: backend/CncApp/CncApp.Application/Dtos/Machines/MachineWithJobsDto.cs
namespace CncApp.Application.Dtos.Machines;

public class MachineWithJobsDto
{
    public int Id { get; set; }
    public string SerialNumber { get; set; } = string.Empty;
    public string ModelNumber { get; set; } = string.Empty;
    public List<MachineJobSummaryDto> Jobs { get; set; } = new();
}

File: frontend/angular/src/app/core/dtos/machines/machine-with-jobs.dto.ts
export interface MachineJobSummaryDto {
  id: number;
  partNumber: string;
  dueDate: string;
  lotNumber: string | null;
  startedDateTime: string | null;
  barsInJob: number;
  barAmountPlanned: number;
  runningShiftId: number | null;
}

export interface MachineWithJobsDto {
  id: number;
  serialNumber: string;
  modelNumber: string;
  jobs: MachineJobSummaryDto[];
}
